app/gui/src/dashboard/components/dashboard/column/LabelsColumn.tsx

```diff
@@ -29,44 +28,62 @@ export default function LabelsColumn(props: column.AssetColumnProps) {
+  const [isOverflowing, setIsOverflowing] = useState(false)
+
+  const labelsList = (item.labels ?? [])
+    .filter((label) => labelsByName.has(label))
+    .map((label) => (
+      <Label
+        key={label}
+        data-testid="asset-label"
+        title={getText('rightClickToRemoveLabel')}
+        color={labelsByName.get(label)?.color ?? FALLBACK_COLOR}
+        active
+        onContextMenu={(event) => {
+          event.preventDefault()
+          event.stopPropagation()
+          const doDelete = () => {
+            unsetModal()
+            const newLabels = item.labels?.filter((oldLabel) => oldLabel !== label) ?? []
+            void backend.associateTag(item.id, newLabels, item.title)
+          }
+          setModal(
+            <ContextMenu aria-label={getText('labelContextMenuLabel')} event={event}>
+              <ContextMenuEntry
+                action="delete"
+                label={getText('deleteLabelShortcut')}
+                doAction={doDelete}
+              />
+            </ContextMenu>,
+          )
+        }}
+        onPress={(event) => {
+          setQuery((oldQuery) =>
+            oldQuery.withToggled('labels', 'negativeLabels', label, event.shiftKey),
+          )
+        }}
+      >
+        {label}
+      </Label>
+    ))
+
   return (
-    <div className="group flex items-center gap-column-items">
-      {(item.labels ?? [])
-        .filter((label) => labelsByName.has(label))
-        .map((label) => (
-          <Label
-            key={label}
-            data-testid="asset-label"
-            title={getText('rightClickToRemoveLabel')}
-            color={labelsByName.get(label)?.color ?? backendModule.COLORS[0]}
-            active
-            onContextMenu={(event) => {
-              event.preventDefault()
-              event.stopPropagation()
-              const doDelete = () => {
-                unsetModal()
-                const newLabels = item.labels?.filter((oldLabel) => oldLabel !== label) ?? []
-                void backend.associateTag(item.id, newLabels, item.title)
-              }
-              setModal(
-                <ContextMenu aria-label={getText('labelContextMenuLabel')} event={event}>
-                  <ContextMenuEntry
-                    action="delete"
-                    label={getText('deleteLabelShortcut')}
-                    doAction={doDelete}
-                  />
-                </ContextMenu>,
-              )
-            }}
-            onPress={(event) => {
-              setQuery((oldQuery) =>
-                oldQuery.withToggled('labels', 'negativeLabels', label, event.shiftKey),
-              )
-            }}
-          >
-            {label}
-          </Label>
-        ))}
+    <div ref={rootRef} className="group relative flex items-center gap-1">
+      <div
+        ref={(el) => {
+          if (!el) {
+            return
+          }
+          setIsOverflowing(el.scrollWidth > el.clientWidth)
```

Comment:
This looks a bit weird. I didn't find any resize logic, and the same time we seems do not react on the amount of labels change?

Reply:
definitely agree, it's semi-intentional because i imagine there would be performance issues, but we can definitely go with a more robust solution with e.g. listening on resize, and on labels change, as long as it's debounced i guess?

Reply:
surprisingly easy to use :D switched to `useMeasureCallback` - but using a ref instead of the rectangle passed to the callback, because I don't know if there's a way to get `scrollWidth`, not just the bounding client rect

let me know if any further changes are desired